h/app.py

```diff
@@ -46,6 +47,32 @@ def create_app(global_config, **settings):
     return config.make_wsgi_app()
 
 
+def publish_after_commit_events(request, response):
+    if request.exception is not None:
+        return
+
+    if len(request.after_commit_events) == 0:
+        return
+
+    with request.tm:
+        request.after_commit_events.publish_all()
+
+
+class EventQueue(collections.deque):
+    def __init__(self, request):
+        self.request = request
+
+    def publish_all(self):
+        while self.has_events():
+            event = self.popleft()
```

Comment:
Oh, and the use of `while` to be able to iterate over the list while modifying it :)